e2e/main.py

```diff
@@ -157,5 +161,152 @@ def call_checker_endpoint(name, param, amount=0):
+        )
+        call_endpoint(ctez["ctez"], "mint_or_burn", (1, 100_000))
+        call_endpoint(ctez["fa12_ctez"], "approve", (checker.context.address, 100_000))
+
+        call_endpoint(
+            checker,
+            "add_liquidity",
+            (100_000, 100_000, 5, int(datetime.now().timestamp()) + 20),
+        )
+
+        burrows = list(range(1, 1001))
+
+        call_bulk(
+            [
+                checker.create_burrow((burrow_id, None)).with_amount(100_000_000)
+                for burrow_id in burrows
+            ],
+            chunk_size=100,
+        )
+
+        # Mint as much as possible from the burrows. All should be identical, so we just query the
+        # first burrow and mint that much kit from all of them.
+        max_mintable_kit = checker.metadata.burrowMaxMintableKit(
+            (self.client.key.public_key_hash(), 1)
+        ).storage_view()
+
+        bulks = []
+        for burrow_no in burrows:
+            bulks.append(
+                checker.touch_burrow((self.client.key.public_key_hash(), burrow_no))
+            )
+            bulks.append(checker.mint_kit(burrow_no, max_mintable_kit))
+
+        call_bulk(bulks, chunk_size=100)
+
+        # Change the index (kits are 100x valuable)
+        #
+        # Keep in mind that we're using a patched checker on tests where the protected index
+        # is much faster to update.
+        call_endpoint(oracle, "update", 100_000_000)
+
+        # Oracle updates lag one touch on checker
+        call_endpoint(checker, "touch", None)
+        call_endpoint(checker, "touch", None)
+        time.sleep(20)
+        call_endpoint(checker, "touch", None)
+
+        # Now burrows should be overburrowed, so we liquidate them all.
+        #
+        # This should use the push_back method of the AVL tree.
+        bulks = []
+        for burrow_no in burrows:
+            bid = (self.client.key.public_key_hash(), burrow_no)
+            bulks.append(checker.touch_burrow(bid))
+            bulks.append(checker.mark_for_liquidation(bid))
+
+        call_bulk(bulks, chunk_size=40)
+
+        # This touch starts a liquidation auction
+        #
+        # This would use the split method of the AVL tree. However, if the entire queue
+        # has less tez than the limit, the 'split' method would trivially return the entire
+        # tree without much effort; so here we should ensure that the liquidation queue has
+        # more tez than the limit (FIXME).
+        call_endpoint(checker, "touch", None)
+
+        # And we place a bid
+
+        ret = checker.metadata.currentLiquidationAuctionMinimumBid().storage_view()
+        auction_id, minimum_bid = ret["contents"], ret["nat_1"]
+        # The return value is supposed to be annotated as "auction_id" and "minimum_bid", I
+        # do not know why we get these names. I think there is an underlying pytezos bug
+        # that we should reproduce and create a bug upstream.
```

Comment:
Ha, that's weird indeed :thinking: Not sure if it's a pytezos bug; it reminds me of #81.